### PR TITLE
test: migrate serde test from mocha to tap

### PR DESCRIPTION
**Pull request by nechaido**
Refs: https://github.com/metarhia/JSTP/issues/145

* separate test cases from tests
* port existing serde tests to tap
* remove old mocha test

### CODE REVIEW

**Comment by nechaido** [comment]
Landed in 0ded6265f3a2ed2358108e59fcd6204a293af72a

File: test/fixtures/testCases/serde.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    value: 1e100,
+    serialized: '1e+100'
+  },
+  {
+    name: 'float with comma',
+    value: 3.14,
+    serialized: '3.14'
+  },
+  {
+    name: 'smallFloat',
```
Comment (belochub) [inline test/fixtures/testCases/serde.js]: Style of this name is inconsistent with overall test naming style in this tests, it should be `'small float'`.

File: test/fixtures/testCases/serde.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    value: 'first\nsecond',
+    serialized: '\'first\\nsecond\''
+  },
+  {
+    name: 'string with single quote',
+    value: 'it\'s',
+    serialized: '\'it\\\'s\''
+  },
+  {
+    name: 'string with \\u',
```
Comment (belochub) [inline test/fixtures/testCases/serde.js]: It'd be better to say `'string with Unicode escape sequences'` here.

**Comment by belochub** [comment]
I am not really happy with the fact, that you landed what seems to be a not completely trivial PR in less then an hour and without getting at least 2 reviews. But that is not the main problem here.
The real issue is the fact that we don't have any rules concerning contributions and landing.
We should probably discuss this topic and create a document. It can even be shared between some of our projects in Metarhia afterwards, especially considering the fact that there will be more people joining us in development of our technological stack in the future.

**Comment by aqrln** [comment]
@belochub for the record, I told @nechaido not to land non-trivial PRs so fast right after he had done that, but +1 for developing strict guidelines and documenting the process.

**Comment by aqrln** [comment]
@nechaido can you please address @belochub's feedback in a follow-up PR?

**Pull request closed without merging**